### PR TITLE
Add more test and refactor existing code

### PR DESCRIPTION
**Pull request by Kacper3331**
@gotar

### CODE REVIEW

File: lib/basket.rb [hunk 1]
```diff
@@ -5,29 +5,34 @@ def initialize(basket=[])
...
+      "You can't buy #{find_product(products, product_id).name}, not enough amount in warehouse"
     end
   end
 
-  def remove_product(products, id, warehouses, quantity)
-    remove_product_from_basket(products, id, warehouses, quantity)
-    puts "Removed from basket: #{find_product(products, id).name} in amount of #{quantity}"
+  def remove_product(product_id, warehouses, quantity)
+    check_basket = basket.find { |basket| basket.id == product_id }
+    if check_basket != nil
```
Comment (wafcio) [inline lib/basket.rb:20]: You can write `!check_backet.nil?` it is more ruby style

File: lib/main.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+require_relative './product'
+require_relative './basket'
+require_relative './warehouse'
```
Comment (wafcio) [inline lib/main.rb:3]: change single quotes to double quotes

File: spec/unit/product_spec.rb [hunk 3]
```diff
@@ -2,46 +2,61 @@
...
+  describe "#id" do
     it "returns id when product is generated" do
-      product_with_price
-      expect(product_with_price.id).to eql(1)
+      expect(product.id).to eql(1)
+    end
+  end
+
+  describe "#show_product" do
+    it "should return information about product" do
```
Comment (wafcio) [inline spec/unit/product_spec.rb:23]: don't use should in test descriptions: http://betterspecs.org/#should

Comment (wafcio) [inline spec/unit/product_spec.rb:23]: please review all tests

File: spec/unit/warehouse_spec.rb [hunk 4]
```diff
@@ -4,20 +4,20 @@
...
+  describe "#product_id " do
+    it "should returns 1 when 1 is given" do
       expect(warehouse.product_id).to eql(1)
     end
   end
 
-  context "#amount " do
-    it "returns 10 when 10 is given" do
+  describe "#amount " do
+    it "should returns 10 when 10 is given" do
```
Comment (gotar) [inline spec/unit/warehouse_spec.rb:20]: do not use should word on the beginning of test description. You should not ask, or please test to work ;]

**Pull request closed without merging**